test/malli/core_test.cljc

```diff
@@ -586,6 +586,74 @@
                                         [false [:sequential int?] #{1 2 3}]
                                         [false [:sequential int?] nil]]
 
+                          "cat" [; same as [:cat {:into nil} ...]
+                                 [true [:cat [:x int?]] '(1)]
+                                 [true [:cat [:x int?]] [1]]
+
+                                 ; aka :cat*
+                                 [true [:cat {:into :any} [:x int?]] '(1)]
+                                 [true [:cat {:into :any} [:x int?]] [1]]
+
+                                 ; aka :catl
+                                 [true [:cat {:into :list} [:x int?]] '(1)]
+                                 [false [:cat {:into :list} [:x int?]] [1]]
+
+                                 ; aka :catv
+                                 [false [:cat {:into :vector} [:x int?]] '(1)]
+                                 [true [:cat {:into :vector} [:x int?]] [1]]
+
+                                 ; concatenated sequences
+                                 [true [:cat [:a [:cat [:ax int?] [:ay int?]]]
+                                             [:b [:cat [:bx int?] [:by int?]]]] [1 2 3 4]]
+
+                                 ; sequence hierarchy
+                                 [true [:cat [:a [:cat {:into :any} [:ax int?] [:ay int?]]]
+                                             [:b [:cat {:into :any} [:bx int?] [:by int?]]]] ['(1 2) [3 4]]]
+
+                                 ; aka :acat
+                                 [true [:cat {:anonymous true} int? int? [:* string?]] [1 2 "a" "b"]]
```

Comment:
currently, both `:or` and `:and` are anonymous by default, should `:cat` be too? Once there is some real use for the named branches, I though of adding variants supporting with names. My initial hunch was to use different schema names, something like:

```clj
;; anonymous
[:or int? string?]

;; named
[:orn [:i int?] [:s str?]]

;; .. or via props
[:or {:named true} [:i int?] [:s str?]]
```

If there are two different things that contribute to schema names, it starts to be awkward:

```clj
;; anonymous
[:cat ...]

;; named branches
[:catn ...]

;; named + vector
[:catnv ...]

;; anonymous + vector
[:catv ...]
```

my 2 cents this morning: the "named branches" would go to name (`:cat` & `:catn`) and the type could be a property. What do you think?

Reply:
The names on `:and`, `:or`, `:cat` and `:alt` do not provide the same kind of information to the user:
- Names on `:and` and `:or` help reporting on which predicates invalidate (`:and`) or validate (`:or`) the data. They are used for indicating **properties** of the data. I guess that the user does not need this information most of the time.
- Names on `:cat` and `:alt` are providing the information on the **identity** of the data. Users would need it most of the time.

To me, it does not look strange for Malli to have by default no names on `:and` and `:or` and have some on `:cat` and `:alt`. Clojure Spec also have a mix of the 2, `s/and` has no name while `s/or`, `s/cat` and `s/alt` have them.

`{:anonymous true/false}` and `{:named: false/true}` encode the same information, I prefer `{:named false/true}` as it is shorter to read and write.